lib/vrf.c

```diff
@@ -964,3 +964,19 @@ int vrf_sockunion_socket(const union sockunion *su, vrf_id_t vrf_id,
 	}
 	return ret;
 }
+
+struct interface *vrf_lookup_interface_by_index(
+			struct route_table *if_table,
+			uint32_t index)
+{
+	struct interface *ifp = NULL;
+
+	if (vrf_is_backend_netns())
+		ifp = if_lookup_by_index_per_route_table(
+				if_table,
+				index);
+	else
+		ifp = if_lookup_by_index(
+				index, VRF_UNKNOWN);
```

Comment:
I think this logic should be moved to `if_lookup_by_index_per_ns()` instead of creating another similar function.

How one should know when to use `vrf_lookup_interface_by_index()`, `if_lookup_by_index()` or `if_lookup_by_index_per_route_table()`? It's extremely confusing.

I think the only long-term solution to fix this chaos would be to get rid of the embryonic support of logical routers (or namespaces) in FRR. This would allow us to make some major simplifications like removing all these weird `if_lookup_*()` functions and rely only on `if_lookup_by_index()` and `if_lookup_by_name()`, which would do the job just fine.

Reply:
Renato, 
I agree to split APIs for now:
- one for zebra : if_lookup_by_index_per_ns()
- one for other daemons : if_lookup_by_index

Obviously, I disagree about proposal of removing support of namespace.
The API to find must encompass both vrf backend.
And looking for an interface across vrfs with an ifindex only is not suitable, since the ifindex may be duplicate across vrf ( in the namespace case).

Reply:
I said that we should consider removing support for logical routers (_lib/logicalrouter.[ch]_ and _zebra/zebra_ns.[ch]_), not removing the netns VRF backend.

Reply:
Removing logical router namespace part ?
In some day someone will offer that capability to have logical router feature.
We should keep maintaining it.

maybe there is other way to do cleanup.

Reply:
Logical routers can be implemented by running multiple instances of FRR on different network namespaces or any other L2 virtualization technology like OpenBSD rdomains. My opinion is that supporting both logical routers and VRFs in FRR is extremely complex and not worth the effort. But this is a conversation that we can have later.